source/server/configuration_impl.h

```diff
@@ -135,6 +159,11 @@ class MainImpl : Logger::Loggable<Logger::Id::config>, public Main {
     return filter_config_factories;
   }
 
+  static std::list<HttpTracerFactory*>& httpTracerFactories() {
+    static std::list<HttpTracerFactory*> http_tracer_factories;
```

Comment:
If we are following the letter of the law here I think this should be a pointer and the list allocated with new